examples/routing_example/server/routes/deprecated_routes/versioned.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { RequestHandler } from '@kbn/core-http-server';
+import type { IRouter } from '@kbn/core/server';
+import { DEPRECATED_ROUTES } from '../../../common';
+
+const createDummyHandler =
+  (version: string): RequestHandler =>
+  (ctx, req, res) => {
+    return res.ok({ body: { result: `API version ${version}.` } });
+  };
+
+export const registerVersionedDeprecatedRoute = (router: IRouter) => {
+  const versionedRoute = router.versioned.get({
+    path: DEPRECATED_ROUTES.VERSIONED_ROUTE,
+    description: 'Routing example plugin deprecated versioned route.',
+    access: 'internal',
```

Comment:
FMI, are you testing anything specific w.r.t. internal routes?

Reply:
no but non-internal routes dont accept custom versioning. they all must be set to a sepcific date